pkg/services/authn/authnimpl/sync/oauth_token_sync.go

```diff
@@ -105,3 +114,30 @@ func getOAuthTokenCacheTTL(t time.Time) time.Duration {
 
 	return ttl
 }
+
+// hasIDTokenExpired checks if the exp field in the ID token has expired. It does not validate the ID token.
+func hasIDTokenExpired(token *login.UserAuth) (bool, error) {
+	if token.OAuthIdToken == "" {
+		return false, nil
+	}
+
+	seperatedIDToken := strings.Split(token.OAuthIdToken, ".")
```

Comment:
Could you please refactor this part to use 'jwt.ParseSigned()` and the `UnsafeClaimsWithoutVerification()` function? In this part the signature validation is not mandatory, because previously it was validated.

Example: https://github.com/grafana/grafana/blob/a97a67fd40dfe2055e2602b0576e1591fee37dc5/pkg/login/social/okta_oauth.go#L56-L64